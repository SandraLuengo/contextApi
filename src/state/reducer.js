import chinesCities from '../fixtures/cities-of-china.json';

const initialState = {
  cities: chinesCities.cities.map(city => ({ ...city, isSelected: false })),
  searching: '',
  isAllChecked: false,
};


const reducer = (state, action) => {
  switch (action.type) {
    case 'SET_CITIES':
      return {
        ...state,
        cities: action.newValue,
      };
    case 'SET_SEARCH':
      return {
        ...state,
        searching: action.newValue,
      };
    case 'SET_IS_ALL_CHECKED':
      return {
        ...state,
        isAllChecked: action.newValue,
      };
    default:
      return state;
  }
};

export { initialState, reducer };

