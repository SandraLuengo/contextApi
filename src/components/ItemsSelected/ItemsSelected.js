import React from 'react';
import { useGlobalState } from '../../state/useGlobalState';
import { selector } from '../functions';
import { SelectedController, CityItem } from '../../components';

const ItemsSelected = ({ className }) => {
  const [{ cities }] = useGlobalState();

  return (
    <div className={className}>
      <SelectedController />
      <div className="itemsSelected">
        {cities.map(city =>
            city.isSelected && (
              <CityItem
                city={city}
                selectorSide
                selector={selector}
                key={city.id}
              />
            ))}
      </div>
    </div>
  );
};

export default ItemsSelected;
