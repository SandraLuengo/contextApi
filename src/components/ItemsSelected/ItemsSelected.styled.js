import styled from 'styled-components';
import ItemsSelected from './ItemsSelected';
import * as style from '../../static/styles/stylesVariables';

export default styled(ItemsSelected).attrs({})`
  height: 100%;
  width: 45%;
  border-left: solid ${style.borderColor} 1.3px;

  .itemsSelected {
    height: 100%;
    max-height: 86vh;
    overflow-y: scroll;
    &::-webkit-scrollbar {
      width: 11px;
      background-color: #f5f5f5;
    }

    &::-webkit-scrollbar-thumb {
      background-color: #7a7a7a;
    }
  }
`;
