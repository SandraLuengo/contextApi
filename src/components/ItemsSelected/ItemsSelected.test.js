import React from 'react';
import renderer from 'react-test-renderer';
import ItemsSelected from './ItemsSelected.styled';
import { Provider } from '../../state/useGlobalState';
import { initialState, reducer } from '../../state/reducer';

const props = { className: 'test' };

it('ItemsSelected component renders correctly', () => {
  const tree = renderer
    .create(<Provider reducer={reducer} initialState={initialState}>
      <ItemsSelected {...props} />
    </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
