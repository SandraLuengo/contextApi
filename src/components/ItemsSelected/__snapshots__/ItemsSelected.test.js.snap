// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsSelected component renders correctly 1`] = `
<div
  className="sc-htoDjs bJacJk test"
>
  <div
    className="sc-ifAKCX euQgUr"
  >
    <p
      className="itemsNumber"
    >
      0
       items
    </p>
    <p
      className="clear"
      onClick={[Function]}
    >
      CLEAR
    </p>
  </div>
  <div
    className="itemsSelected"
  />
</div>
`;
